### DISCUSSION TITLE
Can dledger ganrantee that once a log entry is committed, it will never disappear due to power outage of the all cluster?

### DISCUSSION THREAD

**Discussion by ahan-io** [openmessaging/dledger#108] [Q&A]
It seems like dledger may commit a log entry when the log entry is still in the page cache?  （If I am wrong, let me know please.）

In the worst case, a log entry is commited, but still in page cache, and the whole cluster just get a power outage, the Log Entry is lost?

**Comment by supervate** [comment]
Now there seems only the async flush mechanism, indeed has the problem...

Now async flush is 10ms interval.

<img width="900" alt="image" src="https://github.com/openmessaging/dledger/assets/30173876/3db934a8-e293-433b-8ecc-2bb186b90f54">
